release-notes/7.0/7.0.0/api-diff/Microsoft.NETCore.App/7.0.0_System.IO.md

```diff
@@ -0,0 +1,156 @@
++        public static DateTime GetLastAccessTimeUtc(SafeFileHandle fileHandle);
++        public static DateTime GetLastWriteTime(SafeFileHandle fileHandle);
++        public static DateTime GetLastWriteTimeUtc(SafeFileHandle fileHandle);
++        [UnsupportedOSPlatformAttribute("windows")]
++        public static UnixFileMode GetUnixFileMode(SafeFileHandle fileHandle);
++        [UnsupportedOSPlatformAttribute("windows")]
++        public static UnixFileMode GetUnixFileMode(string path);
++        public static IAsyncEnumerable<string> ReadLinesAsync(string path, Encoding encoding, CancellationToken cancellationToken = default(CancellationToken));
++        public static IAsyncEnumerable<string> ReadLinesAsync(string path, CancellationToken cancellationToken = default(CancellationToken));
++        public static void SetAttributes(SafeFileHandle fileHandle, FileAttributes fileAttributes);
++        public static void SetCreationTime(SafeFileHandle fileHandle, DateTime creationTime);
++        public static void SetCreationTimeUtc(SafeFileHandle fileHandle, DateTime creationTimeUtc);
++        public static void SetLastAccessTime(SafeFileHandle fileHandle, DateTime lastAccessTime);
++        public static void SetLastAccessTimeUtc(SafeFileHandle fileHandle, DateTime lastAccessTimeUtc);
++        public static void SetLastWriteTime(SafeFileHandle fileHandle, DateTime lastWriteTime);
++        public static void SetLastWriteTimeUtc(SafeFileHandle fileHandle, DateTime lastWriteTimeUtc);
++        [UnsupportedOSPlatformAttribute("windows")]
++        public static void SetUnixFileMode(SafeFileHandle fileHandle, UnixFileMode mode);
++        [UnsupportedOSPlatformAttribute("windows")]
++        public static void SetUnixFileMode(string path, UnixFileMode mode);
+     }
+     public sealed class FileInfo : FileSystemInfo {
+-        public override string ToString();
+     }
+     public class FileStream : Stream {
+-        [UnsupportedOSPlatformAttribute("ios")]
+-        [UnsupportedOSPlatformAttribute("macos")]
+-        [UnsupportedOSPlatformAttribute("tvos")]
+-        public virtual void Lock(long position, long length);
++        [UnsupportedOSPlatformAttribute("freebsd")]
++        [UnsupportedOSPlatformAttribute("ios")]
++        [UnsupportedOSPlatformAttribute("macos")]
++        [UnsupportedOSPlatformAttribute("tvos")]
++        public virtual void Lock(long position, long length);
+-        [UnsupportedOSPlatformAttribute("ios")]
+-        [UnsupportedOSPlatformAttribute("macos")]
+-        [UnsupportedOSPlatformAttribute("tvos")]
+-        public virtual void Unlock(long position, long length);
++        [UnsupportedOSPlatformAttribute("freebsd")]
++        [UnsupportedOSPlatformAttribute("ios")]
++        [UnsupportedOSPlatformAttribute("macos")]
++        [UnsupportedOSPlatformAttribute("tvos")]
++        public virtual void Unlock(long position, long length);
```

Comment:
Only 1 attribute was added, this diff suggests that 3 were removed and 4 were added.

```suggestion
+        [UnsupportedOSPlatformAttribute("freebsd")]
        public virtual void Lock(long position, long length);
+        [UnsupportedOSPlatformAttribute("freebsd")]
        public virtual void Unlock(long position, long length);
```

Reply:
Yeah, the api-diff tool is dump in properly diffing attributes differences. It shows everything or nothing. Thanks for the suggested correction.